tools/manifest/sourcefile.py

```diff
@@ -233,7 +233,7 @@ def __init__(self, tests_root, rel_path, url_base, hash=None, contents=None):
         self.url_base = url_base
         self.contents = contents
         self.items_cache = None  # type: Optional[Tuple[Text, List[ManifestItem]]]
-        self._hash = hash
+        self._hash = ensure_str(hash, encoding='ascii') if hash else None
```

Comment:
The mypy annotation makes it look like this should always be Text on input. Why do we end up converting to bytes in Python 2?

Reply:
Indeed, seems a leftover of one the various approaches I took